pennylane/templates/subroutines/trotter.py

```diff
@@ -444,3 +448,366 @@ def compute_decomposition(*args, **kwargs):
+            TrotterizedQfunc(
+                time, 
+                *angles, 
+                qfunc=first_order_expansion, 
+                n=num_trotter_steps, 
+                order=2, 
+                wires=['a', 'b', 'c'], 
+                flip=True,
+            )
+            return qml.state()
+        
+    We can visualize the circuit to see the Suzuki-Trotter product formula being applied:
+            
+        >>> time = 0.1
+        >>> angles = (0.12, -3.45)
+        >>> 
+        >>> print(qml.draw(my_circuit, level=3)(time, angles, num_trotter_steps=1))
+        a: ──RX(0.01)──╭●─╭●──RX(0.01)──┤  State
+        b: ──RY(-0.17)─╰X─╰X──RY(-0.17)─┤  State
+        >>>
+        >>>
+        >>> print(qml.draw(my_circuit, level=3)(time, angles, num_trotter_steps=3))
+        a: ──RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)──┤  State
+        b: ──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)─┤  State
+
+    """
+
+    def __init__(
+        self,
+        time,
+        *trainable_args,
+        qfunc=None,
+        n=1,
+        order=2,
+        reverse=False,
+        name=None,
+        id=None,
+        **non_trainable_kwargs,
+    ):
+        # Enforce the function signature: f(time, arg1, ..., arg_n, wires=[...], kwarg_1, ..., kwarg_n)
+
+        if qfunc is None:
+            raise ValueError("The qfunc must be provided to be trotterized.")
+
+        try:
+            wires = non_trainable_kwargs.pop("wires")
+        except KeyError:
+            wires = trainable_args[-1]
+            trainable_args = trainable_args[:-1]  # exclude the wires from the args
+
+        self._hyperparameters = non_trainable_kwargs
+        self._hyperparameters["n"] = n
+        self._hyperparameters["order"] = order
+        self._hyperparameters["qfunc"] = qfunc
+        self._hyperparameters["reverse"] = reverse
+
+        super().__init__(time, *trainable_args, wires=wires, id=id)
+
+        if name:
+            self._name = name  # Override name if a custom name is provided
+
+    def decomposition(self) -> list[Operator]:
+        """The decomposition"""
+        n = self.hyperparameters["n"]
+        order = self.hyperparameters["order"]
+        qfunc = self.hyperparameters["qfunc"]
+        reverse = self.hyperparameters["reverse"]
+
+        time = self.parameters[0]
+        qfunc_args = self.parameters[1:]
+
+        base_hyper_params = ("n", "order", "qfunc", "reverse")
+        qfunc_kwargs = {k: v for k, v in self.hyperparameters.items() if not k in base_hyper_params}
+
+        decomp = (
+            _recursive_qfunc(
+                time / n, order, qfunc, self.wires, reverse, *qfunc_args, **qfunc_kwargs
+            )
+            * n
+        )
+
+        if qml.QueuingManager.recording():
+            for op in decomp:  # apply operators in reverse order of expression
+                qml.apply(op)
```

Comment:
Should this depend on the `reverse` argument? We don't know the order of `decomp` since it was dependent on `reverse`.